catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/util/EndpointUtilTest.java

```diff
@@ -0,0 +1,204 @@
+    EndpointUtil endpointUtil;
+
+    List<MetacardType> metacardTypeList;
+
+    MetacardType metacardTypeMock;
+
+    CatalogFramework catalogFrameworkMock;
+
+    FilterBuilder filterBuilderMock;
+
+    List<InjectableAttribute> injectableAttributeList;
+
+    InjectableAttribute injectableAttributeMock;
+
+    AttributeRegistry attributeRegistryMock;
+
+    Filter filterMock;
+
+    AttributeBuilder attributeBuilderMock;
+
+    ContextualExpressionBuilder contextualExpressionBuilderMock;
+
+    QueryResponse responseMock;
+
+    Metacard metacardMock;
+
+    Result resultMock;
+
+    @Before
+    public void setUp() throws Exception {
+
+        // mocks
+        metacardTypeList = new ArrayList<>();
+        metacardTypeMock = mock(MetacardType.class);
+        metacardTypeList.add(metacardTypeMock);
+
+        catalogFrameworkMock = mock(CatalogFramework.class);
+
+        filterBuilderMock = mock(FilterBuilder.class);
+
+        injectableAttributeList = new ArrayList<>();
+        injectableAttributeMock = mock(InjectableAttribute.class);
+        injectableAttributeList.add(injectableAttributeMock);
+
+        attributeRegistryMock = mock(AttributeRegistry.class);
+        filterMock = mock(Filter.class);
+        attributeBuilderMock = mock(AttributeBuilder.class);
+        contextualExpressionBuilderMock = mock(ContextualExpressionBuilder.class);
+
+        responseMock = mock(QueryResponse.class);
+        metacardMock = mock(Metacard.class);
+        resultMock = mock(Result.class);
+
+        // when
+        when(filterBuilderMock.attribute(any())).thenReturn(attributeBuilderMock);
+        when(attributeBuilderMock.is()).thenReturn(attributeBuilderMock);
+        when(attributeBuilderMock.like()).thenReturn(contextualExpressionBuilderMock);
+        when(contextualExpressionBuilderMock.text(anyString())).thenReturn(filterMock);
+        when(catalogFrameworkMock.query(any(QueryRequestImpl.class))).thenReturn(responseMock);
+
+        when(resultMock.getMetacard()).thenReturn(metacardMock);
+        when(metacardMock.getId()).thenReturn("MOCK METACARD");
+
+        // constructor
+        endpointUtil = new EndpointUtil(metacardTypeList,
+                catalogFrameworkMock,
+                filterBuilderMock,
+                injectableAttributeList,
+                attributeRegistryMock);
+
+    }
+
+    @Test
+    public void testGetMetacardsByFilterExpectAll()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        String tagFilter = "";
+        int numResults = 100;
+
+        List<Result> resultList = populateResultMockList(numResults);
+        List<Result> emptyList = populateResultMockList(0);
+
+        when(responseMock.getResults()).thenReturn(resultList, resultList, emptyList);
+
+        Map<String, Result> expected = endpointUtil.getMetacardsByFilter(tagFilter);
+
+        assertThat("Should return " + numResults + " results, but returned " + expected.size(),
```

Comment:
❓ We've programmed `responseMock.getResults()` to return 2 lists of 100 results, then an empty one. Shouldn't we be expecting 200 results from the `getMetacardsByFilter()`?
✏️ We could use `assertThat(expected, hasSize(numResults));` instead and not need the message.
✏️ I would also suggest inverting the names of the variables, i.e., set `expected = 100` and `result = endpointUtil.getMetacardByFilter(...)`.